src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -210,6 +207,32 @@ public void AddCompilationRoot(TypeDesc type, string reason)
                     _graph.AddRoot(_factory.ConstructedTypeSymbol(type), reason);
                 }
             }
+
+            public void RootStaticBasesForType(TypeDesc type, string reason)
+            {
+                if (type.IsGenericDefinition)
+                    return;
+
+                MetadataType metadataType = type as MetadataType;
+                if (metadataType != null)
+                {
+                    if (metadataType.ThreadStaticFieldSize > 0)
+                    {
+                        _graph.AddRoot(_factory.TypeThreadStaticsSymbol(metadataType), reason);
+                        _graph.AddRoot(_factory.TypeThreadStaticIndex(metadataType), reason);
```

Comment:
We only need the threadstatic index node. `TypeThreadStaticsSymbol` is an "implementation detail" of the index cell.